platform/security/certificate/security-certificate-generator/src/main/java/org/codice/ddf/security/certificate/generator/CertificateCommand.java

```diff
@@ -18,42 +18,83 @@
 import java.security.cert.X509Certificate;
 import java.util.Arrays;
 
+import javax.annotation.Nullable;
+
 import org.apache.commons.lang.Validate;
 import org.bouncycastle.asn1.x500.RDN;
 import org.bouncycastle.asn1.x500.style.BCStyle;
 
 import ddf.security.SecurityConstants;
 
 public class CertificateCommand {
-
     /**
      * Pass in a string to use as the common name of the certificate to be generated.
-     * Exception thrown if 0 arguments or more than 1 argument.
+     * Exception thrown if 0 arguments or more than 2 argument (or more than 4 when -san is used).
+     * <pre>
+     * Arguments to this program are: <code>[-cn &lt;cn&gt;|-dn &lt;dn&gt;] [-san &lt;tag:name,tag:name,...&gt;]</code>
      *
-     * @param args
+     * where:
+     * &lt;cn&gt; represents a fully qualified common name (e.g. "&lt;FQDN&gt;", where &lt;FQDN&gt; could be something like cluster.yoyo.com)
+     * &lt;dn&gt; represents a distinguished name as a comma-delimited string (e.g. "c=US, st=California, o=Yoyodyne, l=San Narciso, cn=&lt;FQDN&gt;")
+     * &lt;tag:name,tag:name,...&gt; represents optional subject alternative names to be added to the generated certificate
+     *    (e.g. "DNS:&lt;FQDN&gt;,DNS:node1.&lt;FQDN&gt;,DNS:node2.&lt;FQDN&gt;"). The format for subject alternative names
+     *    is similar to the OpenSSL X509 configuration format. Supported tags are:
+     *      email - email subject
+     *      URI - uniformed resource identifier
+     *      RID - registered id
+     *      DNS - hostname
+     *      IP - ip address (V4 or V6)
+     *      dirName - directory name
+     * </pre>
+     *
+     * @param args arguments to the certificate program (see above for description)
      */
     public static void main(String args[]) {
-        if (args.length != 2) {
-            String canonicalName = CertificateCommand.class.getCanonicalName();
-            String exCn = String.format("java %s -cn \"John Smith\"", canonicalName);
+        // try to extract -san if provided
+        int expected = 2;
+        String[] sans = null;
```

Comment:
I usually prefer to avoid `null` when there's another option but this is all internal so no big deal.